pandas/core/arrays/arrow/array.py

```diff
@@ -1533,6 +1533,12 @@ def _reduce(self, name: str, *, skipna: bool = True, **kwargs):
 
         return result.as_py()
 
+    def _reduce_and_wrap(self, name: str, *, skipna: bool = True, kwargs):
```

Comment:
why are you adding another method here? what's wrong with just fixing _reduce?

Reply:
`_reduce` on 1d arrays only returns a scalar and we can't differentiate between scalars from reductions from e.g. numpy.int64 and pandas.Int64() arrays. Reductions that return `pd.NA` are just as bad, because `pd.NA` holds no dtype info.

Also, we can't supply `keepdims` to `_reduce`, because pandas raises when `keepdims` is given as a parameter in the reduction methods.

Reply:
so why don't u just update _reduce?

Reply:
There was a version similar to this that added a keepdims kwd to _reduce and we decided that this was better bc it didn't require a deprecation path for 3rd party EAs

Reply:
_reduce calls other methods, e.g. sum. It's in those methods the failures happen when we give keepdims=True and those methods are public. Do we want to change their signatures (and the ._reduce signature) to include keepdims?

Reply:
If it's just ading the keepdims keyword to `_reduce`, that will be relatively technically easy. It's adding it to `sum` etc. that probably will take more effort. Also note @jbrockmendel comment about deprecation path.

Reply:
Compatibility concerns aside, I think the `keepdims` argument to `_reduce` is the nicer solution. 
But, external EAs don't have this keyword, so that means we would need to add some compatibility code anyway, everywhere we call the `_reduce` method (check if it supports the new keyword, and if not still wrap the result in `np.array([res])` array, just like the current base implementation of `_reduce_and_wrap` does). With that, I am not sure that will be an improvement over the current solution.

Reply:
If a hypothetical EA wanted to do a reduction lazily, that would be much easier with a keepdims keyword than with a _reduce_and_wrap method.  Just a thought, not worth contorting ourselves over a hypothetical EA

Reply:
I'm not opposed to a `keepdims` parameter in `_reduce`, except for the compatibility concerns, but I would like a decision.

One way to address the compatibility concerns could be to introspect the signature of `_reduce` to see if it has a `keepdims` parameter or not. If it does, call `_reduce` with `keepdims=True` when doing dataframe reductions. If it doesn't, call it without a `keepdims` parameter and emit a warning that `keepdims` will become required in the future.